tlf-assemble.Rmd

```diff
@@ -0,0 +1,154 @@
+# Assemble RTF TLFs {#assemble}
+
+## Assemble TLFs to an RTF File
+
+After TLFs are created, we may need to assemble them to one file to simplify review and delivery.
+You can use `r2rtf::rtf_assemble()` to achieve the goal.
+
+```{r, include=FALSE}
+source("common.R")
+```
+
+```{r}
+library(r2rtf)
+```
+
+We use the TLFs generated in the previous chapters as an example to illustrate the workflow.
+
+### List RTF files to assemble
+
+List all RTF files in the `outtable/` directory and/or the `outgraph/` directory.
+
+The R code below lists all `.rtf` files in the output tables in the
+`outtable/` directory and the output graphs in the `outgraph/` directory.
+
+<!-- Note to authors: please modify the code below to find a better way to present this -->
+
+```{r, eval=FALSE, echo=TRUE}
+path <- list(
+  outtable = "tlf/",
+  outgraph = "tlf/"
+)
+
+# For a project the path has been pre-defined
+rtf_table_path <- list.files(path$outtable, pattern = ".rtf")
+rtf_graph_path <- list.files(path$outgraph, pattern = ".rtf")
+
+dput(rtf_table_path)
+#> c("intro-ae1.rtf", "intro-ae2.rtf", "intro-ae3.rtf", "intro-ae5.rtf",
+#> "intro-ae7.rtf", "tbl_disp.rtf", "tbl_pop.rtf", "tlf_ae_summary.rtf",
+#> "tlf_base.rtf", "tlf_eff.rtf", "tlf_eff1.rtf", "tlf_eff2.rtf",
+#> "tlf_km.rtf", "tlf_spec_ae.rtf")
+
+dput(rtf_graph_path)
+#> c("intro-ae1.rtf", "intro-ae2.rtf", "intro-ae3.rtf", "intro-ae5.rtf",
+#> "intro-ae7.rtf", "tbl_disp.rtf", "tbl_pop.rtf", "tlf_ae_summary.rtf",
+#> "tlf_base.rtf", "tlf_eff.rtf", "tlf_eff1.rtf", "tlf_eff2.rtf",
+#> "tlf_km.rtf", "tlf_spec_ae.rtf")
+```
+
+The complete list of `.rtf` files are printed in the Console.
+You can then use this list as a starting point to arrange the order of all files.
+
+An alternative way to get the RTF file list is to read from a spreadsheet
+(e.g., using A&R grid) and work from there.
+
+### Define file order and identify page orientation
+
+Next, we should properly define the order of all TLF files,
+and identify each TLF's page orientation (horizontal or vertical).
+
+The R code below constructs an intermediate data frame `db` with three columns.
+Each TLF file has its respective row to indicate
+
+- `table`: Whether it is table or a plot, `TRUE` or `FALSE`
+- `file`: File name
+- `landscape`: Page orientation, `TRUE` or `FALSE`
+
+<!-- Note to authors: please modify the code below following the last chunk, this is just a placeholder -->
+
+```{r}
+db <- data.frame(
+  file = c(
+    "intro-ae1.rtf",
+    "intro-ae2.rtf",
+    "intro-ae3.rtf",
+    "intro-ae5.rtf",
+    "intro-ae7.rtf",
+    "tbl_disp.rtf",
+    "tbl_pop.rtf",
+    "tlf_ae_summary.rtf",
+    "tlf_base.rtf",
+    "tlf_eff.rtf",
+    "tlf_eff1.rtf",
+    "tlf_eff2.rtf",
+    "tlf_km.rtf",
+    "tlf_spec_ae.rtf"
+  ),
+  table = c(rep(TRUE, 3), FALSE, rep(TRUE, 6), FALSE, rep(TRUE, 2), FALSE),
+  landscape = c(rep(FALSE, 13), rep(TRUE, 1)),
+  stringsAsFactors = FALSE
+)
+
+db
```

Comment:
A minor comment: instead of showing a raw R table, is it better to show a gt table?

Reply:
That would look nice but will also introduce additional 42 packages down the line as hard dependencies to compile the book.